handlers.go

```diff
@@ -3153,6 +3156,75 @@ func (s *Server) handleAddContentsToCollection(c echo.Context, u *User) error {
 	return c.JSON(200, map[string]string{})
 }
 
+// handleCommitCollection godoc
+// @Summary      Produce a CID of the collection contents
+// @Description  This endpoint is used to save the contents in a collection, producing a top-level CID that references all the current CIDs in the collection.
+// @Param        coluuid     path     string  true     "coluuid"
+// @Tags         collections
+// @Produce      json
+// @Success      200  {object}  string
+// @Router       /collections/{collection-id}/commit [post]
+func (s *Server) handleCommitCollection(c echo.Context, u *User) error {
+	colid := c.Param("coluuid")
+
+	var col Collection
+	if err := s.DB.First(&col, "uuid = ? and user_id = ?", colid, u.ID).Error; err != nil {
+		return err
+	}
+
+	contents := []Content{}
+	if err := s.DB.Debug().
+		Model(CollectionRef{}).
+		Where("collection = ?", col.ID).
+		Joins("left join contents on contents.id = collection_refs.content").
+		Select("contents.*").
+		Scan(&contents).Error; err != nil {
+		return err
+	}
+
+	collectionNode := unixfs.EmptyDirNode()
+	for _, c := range contents {
+		collectionNode.AddRawLink(c.Name, &ipld.Link{
+			Size: uint64(c.Size),
+			Cid:  c.Cid.CID,
+		})
+	}
+
+	// update DB with new collection CID
+	col.CID = collectionNode.Cid().String()
+	err := s.DB.Model(Collection{}).Where("id = ?", col.ID).UpdateColumn("c_id", collectionNode.Cid().String()).Error
+	if err != nil {
+		return err
+	}
+
+	// transform listen addresses (/ip/1.2.3.4/tcp/80) in full p2p multiaddresses
+	// e.g. /ip/1.2.3.4/tcp/80/p2p/12D3KooWCVTKbuvrZ9ton6zma5LNhCEeZyuFtxcDzDTmWh2qPtWM
+	fullP2pMultiAddrs := []multiaddr.Multiaddr{}
```

Comment:
if you want to validate like this, best to do this before updating the collection, so we can fail early. Also, why not just pass `[]peer.AddrInfo{{ID: s.Node.Host.ID(), Addrs: s.Node.Host.Addrs()}}`, any need for the validation?

Reply:
I'm not validating, I'm just transforming the listen addresses in a format that's suited for `peer.AddrInfosFromP2pAddrs`